bin/xbps-checkvers/main.c

```diff
@@ -591,18 +476,17 @@ rcv_set_distdir(rcv_t *rcv, const char *distdir)
 }
 
 static bool
-check_reverts(const char *repover, const map_item_t reverts)
+check_reverts(const char *repover, const char *reverts)
 {
 	bool rv = false;
 	char *sreverts, *p;
+	size_t len;
 
-	if (reverts.v.len == 0 || strlen(repover) < 1)
-		return rv;
+	assert(reverts);
+	if (!(len = strlen(reverts)))
+		return false;
 
-	sreverts = calloc(reverts.v.len+1, sizeof(char));
-	assert(sreverts);
-	strncpy(sreverts, reverts.v.s, reverts.v.len);
-	sreverts[reverts.v.len] = '\0';
+	assert((sreverts = strdup(reverts)));
```

Comment:
I would change this to not use assert(), because there's an alloc that could be missed if compiled with -DNDEBUG